### DISCUSSION TITLE
Recover PVs from existing logical volumes

### DISCUSSION THREAD

**Discussion by suleymanakbas91** [topolvm/topolvm#742] [Q&A] [answered]
Hi, 

Is there currently a way to recover/revive PVs from existing logical volumes? Let's say a user installed TopoLVM onto a cluster where there were existing lvs from an earlier installation, and the user wants to use the existing data. If not, would this be a desired/feasible feature?

**Comment by llamerada-jp** [comment, marked as answer]
Hi,

> Is there currently a way to recover/revive PVs from existing logical volumes?

No, there is no way to recover/revive PVs from existing logical volumes. You would need to create new pairs of PV and logical volume,  and manually copy the data between the logical volumes.

> Let's say a user installed TopoLVM onto a cluster where there were existing lvs from an earlier installation, and the user wants to use the existing data. If not, would this be a desired/feasible feature?

It is very difficult.
If at least one of the PVC, PV, or LogicalVolumeCR disappears, only an admin knows which logical volume and which PVC to associate with it.
Even if the admin knows which logical volume to associate with a PVC, there is no way to pass that information to the topolvm-controller, as I checked the [CSI specs](https://github.com/container-storage-interface/spec).
